src/UI/Routing/PrimaryRouteManager.ts

```diff
@@ -154,6 +155,21 @@ export class PrimaryRouteManager implements RouteManager {
     return this.getUrl("CompileDetails", { id: match.params.id });
   }
 
+  getUrlCompileDetailsId(
```

Comment:
How about `getParamsFromUrl` or something like that? 
We are interested in the paramaters (and the `RouteKind`) that will be returned, not the url itself, and the fact that it's for the `CompileDetails` view is already documented with the return type

Reply:
that's fine by me to :) i'll change it